src/lib/utils/wrap.js

```diff
@@ -0,0 +1,11 @@
+export default function wrap(Type, fn, dontCloneWithNoArgs) {
```

Comment:
It's clear enough to me. I'm not exactly sure how one would go about doing what you are doing besides this, while still mostly preserving existing code.